src/instructlab/cli/model/upload.py

```diff
@@ -0,0 +1,249 @@
+        else:
+            self.local_model_path = os.path.join(self.model_dir, self.model)
+            # throw an error if the built path is invalid
+            if not os.path.exists(self.local_model_path):
+                click.secho(
+                    f"Couldn't find model at {self.local_model_path} - are you sure it exists?",
+                    fg="red",
+                )
+                raise click.exceptions.Exit(1)
+
+        # upload either safetensors or gguf, if valid
+        if is_model_safetensors(self.local_model_path):
+            self.upload_safetensors()
+        elif is_model_gguf(self.local_model_path):
+            self.upload_gguf()
+        else:
+            click.secho(
+                f"Local model path {self.local_model_path} is a valid path, but is neither safetensors nor a GGUF - cannot upload to Hugging Face",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+
+    def upload_gguf(self) -> None:
+        try:
+            hf_api = HfApi()
+            resp = hf_api.upload_file(
+                path_or_fileobj=self.local_model_path,
+                path_in_repo=self.model,
+                repo_id=self.upload_dest,
+                revision=self.release,
+                token=self.hf_token,
+            )
+            logger.debug(f"Hugging Face response:\n{resp}")
+        except Exception as exc:
+            click.secho(
+                f"\nUploading GGUF model at {self.local_model_path} failed with the following Hugging Face Hub error:\n{DEFAULT_INDENT}{exc}",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+        click.secho(
+            f"\nUploading GGUF model at {self.local_model_path} succeeded!",
+            fg="green",
+        )
+
+    def upload_safetensors(self) -> None:
+        try:
+            hf_api = HfApi()
+            resp = hf_api.upload_folder(
+                folder_path=self.local_model_path,
+                repo_id=self.upload_dest,
+                revision=self.release,
+                token=self.hf_token,
+            )
+            logger.debug(f"Hugging Face response:\n{resp}")
+        except Exception as exc:
+            click.secho(
+                f"\nUploading safetensors model at {self.local_model_path} failed with the following Hugging Face Hub error:\n{DEFAULT_INDENT}{exc}",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+        click.secho(
+            f"\nUploading safetensors model at {self.local_model_path} succeeded!",
+            fg="green",
+        )
+
+
+@click.command()
+@click.option(
+    "--model",
+    "-m",
+    required=True,
+    help="Local model to upload.",
+)
+@click.option(
+    "--model-dir",
+    default=lambda: DEFAULTS.CHECKPOINTS_DIR,
+    show_default="The default system model location store, located in the data directory.",
+    help="The local directory to upload the model files from.",
+)
```

Comment:
I'm not seeing the benefit of having both here
can we just keep `--model`? best not to have a default IMO cause it might be prone to accidental uploads cause we cant say what checkpoint/other model a user would automatically want pushed out

Reply:
So my thought was it would make it easier/clearer to allow us to specify just the name of a checkpoint rather than users having to pass the absolute path, while also allowing them to pass an absolute path if they prefer

Reply:
To close the loop, I've aligned the scheme used here to match what's in the dev doc - @alinaryan will be replacing this logic across the project with a utility function at a later date